distributed-prover/src/circuits/evm.rs

```diff
@@ -0,0 +1,216 @@
+    // What is <_, G1Affine, Challenge255<_>> in between ::s?
+    // What does this transcript contain?
+    let mut transcript = Blake2bWrite::<_, G1Affine, Challenge255<_>>::init(vec![]);
+
+    let timer = start_timer!(|| "Artifact Deserialization");
+    let general_params = read_params_kzg(CIRCUIT_DEGREE, false);
+    
+    // TODO: check if the EVM circuit has the right syntax
+    let mut pk = read_pk::<EvmCircuit<Fr>>(CIRCUIT_NAME, circuit.params());
+
+    let network_config = read_network_config(CIRCUIT_NAME);
+    let workload_config = read_workload_config(CIRCUIT_NAME);
+    end_timer!(timer);
+
+    let timer = start_timer!(|| "Evaluator configuration");
+    pk.configure_evalutor(workload_config.for_prover(prover_index));
+    end_timer!(timer);
+
+    let timer = start_timer!(|| format!("Prover {} create_proof", prover_index));
+    create_proof_distributed::<
+        KZGCommitmentScheme<Bn256>,
+        ProverSHPLONK<'_, Bn256>,
+        Challenge255<G1Affine>,
+        XorShiftRng,
+        Blake2bWrite<Vec<u8>, G1Affine, Challenge255<G1Affine>>,
+        TestEvmCircuit<Fr>,
+    >(
+        &general_params,
+        &pk,
+        &[circuit],
+        &[&[]],
+        rng,
+        &mut transcript,
+        &network_config,
+        &workload_config,
+        prover_index,
+    ).unwrap();
+    end_timer!(timer);
+}
+
+pub(crate) fn prove_local() {
+    let rng = XorShiftRng::from_seed(RNG_SEED);
+    let circuit = circuit();
+    let mut transcript = Blake2bWrite::<_, G1Affine, Challenge255<_>>::init(vec![]);
+
+    let timer = start_timer!(|| "Artifact deserialization");
+    let general_params = read_params_kzg(CIRCUIT_DEGREE, false);
+    // TODO: check if the EVM circuit has the right syntax
+    let pk = read_pk::<EvmCircuit<Fr>>(CIRCUIT_NAME, circuit.params());
+    end_timer!(timer);
+
+    let timer = start_timer!(|| format!("Prover {} create_proof", 0));
+    create_proof_local::<
+        KZGCommitmentScheme<Bn256>,
+        ProverSHPLONK<'_, Bn256>,
+        Challenge255<G1Affine>,
+        XorShiftRng,
+        Blake2bWrite<Vec<u8>, G1Affine, Challenge255<G1Affine>>,
+        TestEvmCircuit<Fr>,
+    >(
+        &general_params,
+        &pk,
+        &[circuit],
+        &[&[]],
+        rng,
+        &mut transcript,
+    ).unwrap();
+    end_timer!(timer);
+
+    let proof = transcript.finalize();
+    let timer = start_timer!(|| "Artifact serialization");
+    write_proof(CIRCUIT_NAME, &proof);
+    end_timer!(timer);
+}
+
+pub(crate) fn verify() {
+    let timer = start_timer!(|| "Artifact deserialization");
+    let general_params = read_params_kzg(CIRCUIT_DEGREE, false);
+    let verifier_params = read_params_kzg(CIRCUIT_DEGREE, true);
+
+    let vk = read_vk::<EvmCircuit<Fr>>(CIRCUIT_NAME, circuit().params());
+    let proof = read_proof(CIRCUIT_NAME);
+    end_timer!(timer);
+
+    // What does this verifier transcript contain?
```

Comment:
The proof is a transcript and I think the verifier simply initializes a transcript with values in the proof.